CONTRIBUTING.md

```diff
@@ -26,7 +26,54 @@ including the linter (which checks for coding style and warnings, see [Style gui
 These checks are ran automatically using [GitHub Actions](#github-actions). You also need at least
 one approval from a maintainer - after submitting a PR, you can request a review from the top right
 Reviewers menu on the Pull Request page.
+### Branches
+Please use the following structure for creating new development branches:
+```
+[username]/[snake_case_branch_name]
+```
+This will make branches easier to find and search for.<br>
+Examples:
+```
+razvanra2/add_authoring_policy
+luigi/dev_eng_improv
+auditore/fix_md_typo
+torvalds/android_speedups
+```
+### Merging
+When developing a new feature or working on a bug, your pull request will end up containing fixup commits (commits that change the same line of code repeatedly) or too fine-grained commits. An issue that can arise from this is that the main branch history will become poluted with unnecessary commits. To avoid it, we implement and enforce a squash policy.
+All commits that are merged intro the main development branch have to be squashed ahead of the merge.
+You can do so by pressing "squash and merge", or, alternetively, following the generic local squash routine outlined bellow:
+```
+git checkout your_branch_name
+git rebase -i HEAD~n
+# n is normally the number of commits in the pull request.
+# Set commits (except the one in the first line) from 'pick' to 'squash', save and quit.
+# On the next screen, edit/refine commit messages.
+# Save and quit.
+git push -f # (force push to GitHub)
+```
+Please update the resulting commit message, if needed. It should read as a coherent message. In most cases, this means not just listing the interim commits.
```

Comment:
The default message when you do it from GitHub is something like:
```
PR title (#pr_number)

* Commit message 1

* Commit message 2
```

Maybe we could suggest that?

Reply:
People would have to manually go back and copy-paste their commit messages. It'd be tedious.
A suggestive message if this is not done when merging from the github gui should be enough.

Reply:
Fair point. I'd still enforce specifying the PR number though, that way you can click it and see more info about the commit from the PR.